### PR TITLE
Fixing miscellaneous issues

### LINKED ISSUE #70: When a command fails it will still be added to the cooldown list for the user
This is because most commands just `return` as if the command ran successfully.

### LINKED ISSUE #71: []clear 2 only deletes 1 message

### LINKED ISSUE #72: If a message contains plain text and codeblock in same message, the plain text is deleted

### LINKED ISSUE #74: Bad way to handle configuration files
I'll make template files and automatic template copying if files don't exist.
This will (hopefully) make merging configuration files painless.

### PR DESCRIPTION
**Pull request by MetalBreaker**
Removed unnecessary user mentions.
Closes #72.
Closes #71.
Exceptions are thrown in commands now instead of returning like nothing happened. 
Closes #70.
Closes #74. Templates are now used for configuration files.
If a command errors out, the bot will delete the error message eventually.
Cleaned up the message deletion code. It doesn't need an await as it needs to execute asynchronously. The _ is for disposing of it and a warning. 
Added cooldowns.json to .gitignore.

### CODE REVIEW

File: BrackeysBot.cs [hunk 1]
```diff
@@ -150,27 +161,23 @@ private async Task HandleCommand(SocketMessage s)
...
+                if (!cooldownExpired)
                 {
                     TimeSpan ts = GetTimeUntilCooldownHasExpired (executedCommand.Name.ToLower (), s.Author as IGuildUser, sameParamCommand, parameters);
 
                     if (executedCommand.Name.ToLower () == "thanks")
                     {
                         Embed eb = new EmbedBuilder ()
                             .WithTitle ("You can't thank that user yet")
-                            .WithDescription ($"{s.Author.Mention}, you can't thank that user yet. Please wait {ts.Hours} hours, {ts.Minutes} minutes and {ts.Seconds} seconds.")
+                            .WithDescription ($"{((IGuildUser)s.Author).GetDisplayName()}, you can't thank that user yet. Please wait {ts.Hours} hours, {ts.Minutes} minutes and {ts.Seconds} seconds.")
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Why are you replacing a mention just with their name? Any particular reason?

File: BrackeysBot.cs [hunk 2]
```diff
@@ -180,7 +187,7 @@ private async Task HandleCommand(SocketMessage s)
                     {
                         Embed eb = new EmbedBuilder ()
                             .WithTitle ("Cooldown hasn't expired yet")
-                            .WithDescription ($"{s.Author.Mention}, you can't run this command yet. Please wait {ts.Hours} hours, {ts.Minutes} minutes and {ts.Seconds} seconds.")
+                            .WithDescription ($"{((IGuildUser)s.Author).GetDisplayName()}, you can't run this command yet. Please wait {ts.Hours} hours, {ts.Minutes} minutes and {ts.Seconds} seconds.")
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Same here.

**Comment by MetalBreaker** [comment]
Yeah, I was replacing mentions with users' display names so as to not bother users that much.
Though, you are right. If the bot's saying ", you blah blah", there should be a mention.

File: Commands/HasteCommand.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public async Task ModHasteMessage(ulong messageId)
             string content = RemoveCodeblockFormat(message.Content);
             string url = await HasteMessage(content);
 
-            await ReplyAsync($"Message by { message.Author.Mention } was hasted to { url }.");
+            await ReplyAsync($"Message by { ((IGuildUser) message.Author).GetDisplayName() } was pasted to { url }.");
```
Comment (CodeMyst) [inline Commands/HasteCommand.cs]: Replace display name with mention.

File: Commands/HasteCommand.cs [hunk 4]
```diff
@@ -63,7 +63,7 @@ public static async Task HasteIfMassiveCodeblock (IMessage message)
                 string code = RemoveCodeblockFormat(content);
                 string url = await HasteMessage(code);
 
-                await message.Channel.SendMessageAsync($"Hastebin created in place of massive codeblock by { message.Author.Mention }: { url }");
+                await message.Channel.SendMessageAsync($"Hastebin created in place of massive codeblock by { ((IGuildUser) message.Author).GetDisplayName() }: { url }");
```
Comment (CodeMyst) [inline Commands/HasteCommand.cs]: Replace display name with mention.

File: Commands/PasteCommand.cs [hunk 5]
```diff
@@ -58,10 +58,11 @@ public class PasteCommand : ModuleBase
...
         {
             var message = await Context.Channel.GetMessageAsync(messageId);
-            string content = RemoveCodeblockFormat(message.Content, out string syntax);
-            string url = await PasteMessage(content, syntax);
+            var messageContent = message.Content;
+            RemoveCodeblockFormat(ref messageContent, out string syntax);
+            string url = await PasteMessage(messageContent, syntax);
 
-            await ReplyAsync($"Message by { message.Author.Mention } was hasted to { url }.");
+            await ReplyAsync($"Message by { ((IGuildUser) message.Author).GetDisplayName() } was pasted to { url }.");
```
Comment (CodeMyst) [inline Commands/PasteCommand.cs]: Replace display name with mention.

File: Commands/PasteCommand.cs [hunk 6]
```diff
@@ -129,10 +131,10 @@ public static async Task PasteIfMassiveCodeblock(IMessage message)
             string content = message.Content;
             if (HasCodeblockFormat(content) && content.Length >= MASSIVE_THRESHOLD)
             {
-                string code = RemoveCodeblockFormat(content, out string syntax);
-                string url = await PasteMessage(code, syntax);
+                RemoveCodeblockFormat(ref content, out string syntax);
+                string url = await PasteMessage(content, syntax);
 
-                await message.Channel.SendMessageAsync($"Paste created in place of massive codeblock by { message.Author.Mention }: <{ url }>");
+                await message.Channel.SendMessageAsync($"Paste created in place of massive codeblock by { ((IGuildUser)message.Author).GetDisplayName() }: <{ url }>");
```
Comment (CodeMyst) [inline Commands/PasteCommand.cs]: Replace display name with mention.

**Comment by MetalBreaker** [comment]
I don't see a reason for that.
A user shouldn't be mentioned in that case..

**Comment by CodeMyst** [comment]
Mentions in an embed don't actually ping the user.

**Comment by CodeMyst** [comment]
Sorry, my bad.

**Comment by CodeMyst** [comment]
I feel like you should have used mentions when someone's codeblock gets "pasted". It clarifies the exact user (similar user names), and notifies the original poster that their codeblock was replaced by a link to a `paste.ee` link.

**Comment by MetalBreaker** [comment]
Yeah. Now that I think about it, that makes sense. Pushed a commit.

**Pull request merged**